src/test/regress/expected/multi_join_order_additional.out

```diff
@@ -168,6 +168,30 @@ LOG:  join order: [ "orders_hash" ][ single range partition join "customer_appen
          explain statements for distributed queries are not enabled
 (3 rows)
 
+-- Validate a 4 way join that could be done locally is planned as such by the logical
+-- planner. It used to be planned as a repartition join due to no 1 table being directly
+-- joined to all other tables, but instead follows a chain.
+EXPLAIN SELECT count(*)
```

Comment:
For reference, and feel free to add to tests, I've played with such queries - changing some columns and making them cartesian products and/or dual/single repartitions. All looks good.
```SQL
SELECT coordinator_plan($$ EXPLAIN SELECT count(*) FROM users_table u1, users_table u2, users_table u3, users_table u4, users_table u5, users_table u6, users_table u7, users_table u8 WHERE u1.user_id = u2.user_id AND u1.user_id = u3.user_id AND u1.user_id = u4.user_id AND u1.user_id = u5.user_id AND u1.user_id = u6.user_id AND u1.user_id = u7.user_id AND u1.user_id = u8.user_id;$$);


SELECT coordinator_plan($$ EXPLAIN SELECT count(*) FROM users_table u1, users_table u2, users_table u3, users_table u4, users_table u5, users_table u6, users_table u7, users_table u8 WHERE u8.user_id = u2.user_id AND u8.user_id = u3.user_id AND u8.user_id = u4.user_id AND u8.user_id = u5.user_id AND u8.user_id = u6.user_id AND u8.user_id = u7.user_id AND u1.user_id = u8.user_id;$$);

SELECT coordinator_plan($$ EXPLAIN SELECT count(*) FROM users_table u1, users_table u2, users_table u3, users_table u4, users_table u5, users_table u6, users_table u7, users_table u8 WHERE u1.user_id = u2.user_id AND u2.user_id = u3.user_id AND u3.user_id = u4.user_id AND u4.user_id = u5.user_id AND u5.user_id = u6.user_id AND u6.user_id = u7.user_id AND u7.user_id = u8.user_id;$$);

```